tests/async_test_class_client/http.py

```diff
@@ -95,41 +95,59 @@ def handle_response(func, response):
 }
 
-auth_key = c.get_bearer_token()
-client_headers = c.additional_headers()
-client_headers.update(Authorization=f"Bearer {auth_key}")
-client = httpx.AsyncClient(headers=client_headers)
 
-
-async def get(url: str, headers: typing.Optional[dict] = None) -> httpx.Response:
-    """Issue an HTTP GET request"""
-    if headers:
-        client_headers.update(headers)
-    async with httpx.AsyncClient(headers=client_headers) as async_client:
-        return await async_client.get(parse_url(url))
-
-
-async def post(url: str, data: dict, headers: typing.Optional[dict] = None) -> httpx.Response:
-    """Issue an HTTP POST request"""
-    if headers:
-        client_headers.update(headers)
-    json_data = json.loads(json.dumps(data, default=json_serializer))
-    async with httpx.AsyncClient(headers=client_headers) as async_client:
-        return await async_client.post(parse_url(url), json=json_data)
-
-
-async def put(url: str, data: dict, headers: typing.Optional[dict] = None) -> httpx.Response:
-    """Issue an HTTP PUT request"""
-    if headers:
-        client_headers.update(headers)
-    json_data = json.loads(json.dumps(data, default=json_serializer))
-    async with httpx.AsyncClient(headers=client_headers) as async_client:
-        return await async_client.put(parse_url(url), json=json_data)
-
-
-async def delete(url: str, headers: typing.Optional[dict] = None) -> httpx.Response:
-    """Issue an HTTP DELETE request"""
-    if headers:
-        client_headers.update(headers)
-    async with httpx.AsyncClient(headers=client_headers) as async_client:
-        return await async_client.delete(parse_url(url))
+class HTTPClient:
+    """HTTP client wrapper that manages the httpx client with configuration."""
+
+    def __init__(self, config: c.Config):
+        self.config = config
+        self._client: typing.Optional[httpx.AsyncClient] = None
+
+    def _get_client(self) -> httpx.AsyncClient:
+        """Get or create the httpx client with current configuration."""
+        if self._client is None:
+            client_headers = self.config.additional_headers.copy()
+            client_headers.update(Authorization=f"Bearer {self.config.bearer_token}")
+            self._client = httpx.AsyncClient(headers=client_headers)
+        return self._client
+
+    def _get_headers(self, additional_headers: typing.Optional[dict] = None) -> dict:
+        """Get headers for a request, merging config and request-specific headers."""
+        headers = self.config.additional_headers.copy()
+        headers.update(Authorization=f"Bearer {self.config.bearer_token}")
+        if additional_headers:
+            headers.update(additional_headers)
+        return headers
+
+    async def get(self, url: str, headers: typing.Optional[dict] = None) -> httpx.Response:
+        """Issue an HTTP GET request"""
+        request_headers = self._get_headers(headers)
+        client = self._get_client()
```

Comment:
I think something is wrong with the asynchronous client here. The diff suggests we should use it as a context manager like `async with` - check the docs for https and make sure we are using it correctly and then adjust based on recommendations.